### ISSUE TITLE
jelly should not import the deprecated sets module

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8096]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8096|
|Type|enhancement|
|Created|2015-11-07 13:27:35Z|
|Branch|https://github.com/twisted/twisted/tree/jelly-sets-8096|

This makes the following warning:

```
/Users/red/code/twgit/twisted/spread/jelly.py:92: DeprecationWarning: the sets module is deprecated
  import sets as _sets
```

<details><summary>Searchable metadata</summary>

```
trac-id__8096 8096
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_jelly_sets_8096 branches/jelly-sets-8096
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1446902855537522 1446902855537522
changetime__1447413951502783 1447413951502783
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [46105]) Branching to jelly-sets-8096.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Looks good now. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Looks good.

Can we use this filter ?
```
warnings.simplefilter('ignore', DeprecationWarning)
```

Please merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46182]) Merge jelly-sets-8096: Make Jelly not raise a DeprecationWarning with its backwards compatible use of sets

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8096](https://github.com/twisted/twisted/issues/8096)
